lib/saml2.coffee

```diff
@@ -595,6 +597,32 @@ module.exports.ServiceProvider =
                 return cb_wf new SAMLError("SAML Response was not success!", {status: get_status(saml_response)})
 
               response.type = 'authn_response'
+
+              conditions = saml_response.getElementsByTagNameNS(XMLNS.SAML, 'Conditions')[0]
+              if conditions?
+                if options.ignore_timing != true
+                  for attribute in conditions.attributes
+                    condition = attribute.name.toLowerCase()
+                    if condition == 'notbefore' and Date.parse(attribute.value) > Date.now()
+                      return cb_wf new SAMLError('SAML Response is not yet valid', {NotBefore: attribute.value})
+                    if condition == 'notonorafter' and Date.parse(attribute.value) <= Date.now()
+                      return cb_wf new SAMLError('SAML Response is no longer valid', {NotOnOrAfter: attribute.value})
+
+                audience_restriction = conditions.getElementsByTagNameNS(XMLNS.SAML, 'AudienceRestriction')[0]
+                audiences = audience_restriction?.getElementsByTagNameNS(XMLNS.SAML, 'Audience')
+                if audiences?.length > 0
+                  validAudience = false
+                  for audience in audiences
```

Comment:
It looks like they're already including underscore .. this feels a lot like an underscore _.find -- `validAudience = _.find(ye predicate)` it'd simplify the code a little bit for the reader and modifier, and I think have all of the same performance characteristics. Would that be idiomatic within this codebase? Minor thing, feel free to brush it off if there are good reasons to do it this way.

Reply:
Thanks @brettkiefer — it is better with `_.find`. I've pushed a fixup for when you're ready to re-review!